tests/temperature_test.cpp

```diff
@@ -155,64 +169,75 @@ TEST_CASE( "Rate of temperature change" )
         // 2x cooked meat (-20 C) warming in 20 C enviroment for ~190 minutes
         // 0 min: both at -20 C and frozen
-        // 10 min: meat 1 at -5.2 C
-        // 20 min: meat 1 step
-        // 30 min: meat 1 step
-        // 50 min: Meat 1 at 0 C an frozen
-        // 140 min: Meat 1 and 2 at 0 C not fozen.
+        // 11 min: meat 1 at -5.2 C
+        // 22 min: meat 1 step
+        // 33 min: meat 1 step
+        // 53 min: Meat 1, 2 at 0 C an frozen
+        // 143 min: Meat 1 and 2 at 0 C not fozen.
         //          Both have same energy (within rounding error)
-        // 150 min: meat 1 step
-        // 170 min: meat 1 step
-        // 190 min: Meat 1 and 2 at 13.3 C
+        // 154 min: meat 1 step
+        // 174 min: meat 1 step
+        // 194 min: Meat 1 and 2 at 13.3 C
 
         item meat1( "meat_cooked" );
         item meat2( "meat_cooked" );
 
-        meat1.update_temp( -4, 1 );
-        meat2.update_temp( -4, 1 );
+        meat1.process_temperature_rot( -4, 1, tripoint( 0, 0, 0 ), nullptr );
+        meat2.process_temperature_rot( -4, 1, tripoint( 0, 0, 0 ), nullptr );
 
         // -20 C
         CHECK( is_nearly( meat1.temperature, 253.15 * 100000 ) );
         CHECK( meat1.item_tags.count( "FROZEN" ) );
 
 
-        calendar::turn = to_turn<int>( calendar::turn + 101_turns );
-        meat1.update_temp( 68, 1 );
+        calendar::turn = to_turn<int>( calendar::turn + 11_minutes );
+        meat1.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
         // -5.2 C
         CHECK( is_nearly( meat1.temperature, 26789608 ) );
 
-        calendar::turn = to_turn<int>( calendar::turn + 101_turns );
-        meat1.update_temp( 68, 1 );
-        calendar::turn = to_turn<int>( calendar::turn + 101_turns );
-        meat1.update_temp( 68, 1 );
+        calendar::turn = to_turn<int>( calendar::turn + 11_minutes );
+        meat1.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
+
+        calendar::turn = to_turn<int>( calendar::turn + 11_minutes );
+        meat1.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
+
+        calendar::turn = to_turn<int>( calendar::turn + 20_minutes );
+        meat1.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
+        meat2.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
 
-        calendar::turn = to_turn<int>( calendar::turn + 197_turns );
-        meat1.update_temp( 68, 1 );
         // 0C
+        // same temp
         // frozen
         CHECK( is_nearly( meat1.temperature, 27315000 ) );
+        CHECK( is_nearly( meat2.temperature, meat1.temperature ) );
         CHECK( meat1.item_tags.count( "FROZEN" ) );
+        CHECK( meat2.item_tags.count( "FROZEN" ) );
+
+        calendar::turn = to_turn<int>( calendar::turn + 45_minutes );
+        meat1.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
+        meat2.process_temperature_rot( 68, 1, tripoint( 0, 0, 0 ), nullptr );
+
```

Comment:
Why following three lines are duplicates of previous three lines?

Reply:
First line moves time forwards.
The two next lines claculate the item temperature at the new time point.

The time delta between calculations needs to be less than 1 hour or the code will start calculating past temperature using weather and the result is completely different.